velox/expression/tests/SignatureBinderTest.cpp

```diff
@@ -40,6 +40,16 @@ void assertCannotResolve(
   ASSERT_FALSE(binder.tryBind());
 }
 
+std::string getResultScale(std::string precision, std::string scale) {
```

Comment:
This function is very specific to the only place where it is used, but has generic name. Let's remove it and re-write the call-site using additional variables, e.g.

```
integerVariable("c_precision", "a_precision + b_precision + 1")
integerVariable("c_scale",  "a_scale + b_scale")
integerVariable("r_scale", "c_precision <= 38 ? ...")
```

Reply:
> We cannot introduce c_precision, will throw exception:

This looks like a bug. Would be nice to fix.

Reply:
This is a constraint, I'm not sure if it is essential.

Reply:
We cannot just remove this check because we do want to catch cases where a variable is defined but never used. However, this check needs to be enhanced to not fire up when variable is used like in the case above.